locals.tf

```diff
@@ -139,9 +139,12 @@ locals {
   infrastructure_ecs_cluster_termination_sns_topic_name            = "${local.resource_prefix}-infrastructure-ecs-cluster-termination"
   infrastructure_ecs_cluster_min_size                              = var.infrastructure_ecs_cluster_min_size
   infrastructure_ecs_cluster_max_size                              = var.infrastructure_ecs_cluster_max_size
-  infrastructure_ecs_cluster_max_instance_lifetime                 = var.infrastructure_ecs_cluster_max_instance_lifetime
-  infrastructure_ecs_cluster_autoscaling_time_based_max            = toset(var.infrastructure_ecs_cluster_autoscaling_time_based_max)
-  infrastructure_ecs_cluster_autoscaling_time_based_min            = toset(var.infrastructure_ecs_cluster_autoscaling_time_based_min)
+  infrastructure_ecs_cluster_allow_kms_encryption = local.infrastructure_kms_encryption && (
+    local.enable_infrastructure_vpc_transfer_s3_bucket
+  )
```

Comment:
You shouldn't need the `()` here. It can be a one-liner `local.foo == local.bar`

Reply:
Yep, I was future proofing that condition - As there may be other reasons we may want to 'allow_kms_encryption'
Will make for a clearer diff :)

Reply:
eg. We'll be able to just add:
```
... (
  local.enable_infrastructure_vpc_transfer_s3_bucket ||
  local.something_else
)
```

Reply:
In that case we can use 
```hcl
alltrue([
    local.mything,
    local.mysecondthing
])
```
instead of chaining `&&`

Reply:
we could also write it as:
```
  infrastructure_ecs_cluster_allow_kms_encryption = local.infrastructure_kms_encryption && contains([
    local.enable_infrastructure_vpc_transfer_s3_bucket
  ], true)
```

Reply:
Clearly a few different options :) Pick your poison

Reply:
Ah yep, we can use `anytrue` - I'll make that change so that it's clearer

Reply:
since trailign commans work i agree that anytrue looks much better and will make for clearer diffs.